src/controllers/TaskController.tsx

```diff
@@ -34,8 +48,19 @@ export const addTask = async (task: Database['public']['Tables']['Tasks']['Inser
     return res.data.task_id;
 }
 
-export const updateTask = async () => {
-    alert("You have asked TaskController to update a task.")
+export const updateTask = async (task: Database['public']['Tables']['Tasks']['Update'], task_id: number) => {
+    const res = await supabase
```

Comment:
What happens if there's no `task_id` given? Could you do a quick if check, similar to `updateProperty` in `PropertyController.tsx`?

Reply:
its really not liking this because then I have to throw the logic in an if statement which makes task_id potentially undefined, and then the taskswithtags stuff gets mad

Reply:
alright well if it works it works